### PR TITLE
ci: Pass tests under macOS

### LINKED ISSUE #294: Add CI for MacOS
### Actual

Previous job is failing with `3.5.1` with:

```yml
  test-macos:
    name: Test on fish on MacOS
    runs-on: macos-latest
    steps:
      - uses: actions/checkout@v2
      - name: Install Fish
        run: brew install fish
      - name: Install Fisher > Fishtape > Pure
        run: curl -sL https://git.io/fisher | source && fisher install jorgebucaran/fisher jorgebucaran/fishtape ./
        shell: fish {0}
      - name: Test Pure
        run: fishtape tests/*.test.fish
        shell: fish {0}
```

### Expected 

To have successful CI job for MacOS and latest Fish version

### PR DESCRIPTION
**Pull request by hyperupcall**
**Related**: Fixes #294

## Description

This fixes the tests so macOS CI passes. There were two main issues:
- [osx wc printing unexpected spaces](https://stackoverflow.com/questions/30927590/wc-on-osx-return-includes-spaces)
  - trimmed the spaces with tr
- `set -U` not working on macOS (currently fish version 3.7)
  - to fix this, erasing the variable before setting it somehow made it work

## First contribution?

Check the [:+1: Contribute][contribute] section and the [contributing guide][contributing].

## Acceptance Checks

* [ ] documentation is up-to-date:
  * [ ] [features list][features] ;
  * [ ] [Prompt Symbol][symbol] ;
  * [ ] [🔌 Features' Flags][feature-flag] ;
* [ ] feature flag is present in `conf.d/pure.fish` ;
* [ ] symbol is present in `conf.d/pure.fish` ;
* [x] tests are passing (I can help you :hugs: ):
  * [ ] config are tested (cf. [tests/_pure.test.fish][config-test]) ;
  * [ ] feature is covered ;
* [ ] customization is available ;
* [x] feature is implemented.

[config-test]: tests/_pure.test.fish
[contribute]: /pure-fish/pure/#1-contribute
[contributing]: CONTRIBUTING.md
[feature-flag]: /pure-fish/pure/#-features-flags
[symbol]: /pure-fish/pure/#prompt-symbol
[features]: /pure-fish/pure/#features

### CODE REVIEW

File: tests/_pure_is_inside_container.test.fish [hunk 1]
```diff
@@ -54,11 +54,15 @@ end
...
+@test "_pure_is_inside_container: false for Github Action" (
     set --universal pure_enable_container_detection true
 
     _pure_is_inside_container
-) $status -eq $SUCCESS
+) $status -eq (if test (uname) = Darwin
+    echo $FAILURE
+else
+    echo $SUCCESS
+end)
```
Comment (edouard-lopez) [inline tests/_pure_is_inside_container.test.fish:65]: Please use a dedicated test instead: 

```fish
if test (uname -s) != Darwin
	cleanup_spy
	@test "_pure_is_inside_container: true for Github Action on Ubuntu" (
	    set --universal pure_enable_container_detection true
	
	    _pure_is_inside_container
	) $status -eq $SUCCESS
	end
end

if test (uname -s) = Darwin
	cleanup_spy
	@test "_pure_is_inside_container: false for Github Action on MacOS" (
	    set --universal pure_enable_container_detection true
	
	    _pure_is_inside_container
	) $status -eq $FAILURE
	end
end
```

**Comment by hyperupcall** [comment]
@edouard-lopez I've updated the tests to use a dedicated container test for macOS and Linux, but I don't think it's possible to specify a particular version of Fish to install with Homebrew. Usually, only the latest versions are supported. Some applications like postgres do have multiple versions and some applications do have custom taps with multiple versions, but that does not seem to be the case for Fish. Maybe that could be fixed by someone a follow-up PR?

File: tests/_pure_is_inside_container.test.fish [hunk 2]
```diff
@@ -53,12 +53,25 @@ end
...
+if test (uname -s) != Darwin
+    cleanup_spy
+    @test "_pure_is_inside_container: true for Github Action on Ubuntu" (
+        set --universal pure_enable_container_detection true
 
-    _pure_is_inside_container
-) $status -eq $SUCCESS
+        _pure_is_inside_container
+    ) $status -eq $SUCCESS
+    end
```
Comment (edouard-lopez) [inline tests/_pure_is_inside_container.test.fish:63]: I introduce an extraneous `end` that causes the [tests to fail](https://github.com/pure-fish/pure/actions/runs/7635299204/job/20807724766?pr=340#step:5:104)
```suggestion
```

File: tests/_pure_is_inside_container.test.fish [hunk 3]
```diff
@@ -53,12 +53,25 @@ end
...
+end
+
+if test (uname -s) = Darwin
+    cleanup_spy
+    @test "_pure_is_inside_container: false for Github Action on MacOS" (
+        set --universal pure_enable_container_detection true
+
+	    _pure_is_inside_container
+    ) $status -eq $FAILURE
+    end
```
Comment (edouard-lopez) [inline tests/_pure_is_inside_container.test.fish:73]: Same here
```suggestion
```

**Review by edouard-lopez — CHANGES_REQUESTED** [review]
I introduced a regression with the code I suggested that added an extra `end` on each block. Could you update and then rebase on master ?

**Comment by edouard-lopez** [comment]
There is an issue with `not ok 97 _pure_is_inside_container: true for Github Action on Ubuntu`. I will have a look

**Comment by edouard-lopez** [comment]
The failure was due to error in the test file (not cleaning a mocked method), I will fix in #347

**Pull request merged**